api/data_refinery_api/views.py

```diff
@@ -485,3 +490,45 @@ def get(self, request, format=None):
         data['processor_jobs']['average_time'] = ProcessorJob.objects.filter(start_time__isnull=False, end_time__isnull=False).aggregate(average_time=Avg(F('end_time') - F('start_time')))['average_time']
 
         return Response(data)
+
+###
+# Transcriptome Indices
+###
+
+class TranscriptomeIndexDetail(APIView):
+    """
+    Retrieve the S3 URL to an OrganismIndex.
+    """
+    def get_s3_url(self, organism_name, transcription_length):
+        """
+        Gets the S3 url associated with the organism and length.
+        Organism must be specified in underscore-delimited uppercase, i.e. "GALLUS_GALLUS".
+        Length must either be "long" or "short"
+        """
+
+        if ORGANISM_INDEX_BUCKET == "NO_BUCKET":
+            raise Http404
+
+        transcription_length = "TRANSCRIPTOME_" + transcription_length.upper()
+
+        try:
+            organism_index = (OrganismIndex.public_objects.exclude(s3_key__exact="")
```

Comment:
I would suggest moving this functionality onto the Organism object itself as get_index, in fact I'd even take that a step further and add a get_index_s3_url to move all of this model logic off of the API view function.